### DISCUSSION TITLE
Linux & PiP (Picture in Picture) - Always on Top

### DISCUSSION THREAD

**Discussion by o-data** [FreeTubeApp/FreeTube#5599] [Q&A] [answered]
Hi there,

Is it Freetube or the Linus distro that determine this issue? I'm using PopOS (distro) and when I use the PiP feature the screen is not Always on Top, it get overlap by other program.

Is it because Freetube PiP not support Linux to be always on top? Or it because the PopOS/Linux doesn't support Always on Top, I can use a Shortcut to force the window to be On Top but it will be bugged when I play a different video.

Anyone having issue using PiP with Linux?

I'm a Windows user and there is no issue here, want to give Linux a try.

**Comment by KuteKimo** [comment]
I also want to know, use ubuntu and it happens the same thing, I hope you can solve it or if you have something you can do now

**Comment by absidue** [comment, marked as answer]
From the research I've done it doesn't seem like Wayland has a standard way for applications to set themselves as always on top, some implementations have custom protocol extensions but for everything else your best bet is to manually set the window to always be on top through your window manager.

**Comment by o-data** [comment]
Thanks @absidue. Unfortunately that doesn't work as well. But at least I know if it not just me. Hopefully future update or patch can fix it.

For anyone who following: https://github.com/pop-os/cosmic-epoch/issues/1073

I'm rather surprise not many people see this issue, people don't use PiP?
Same issue for other software too, Firefox and Floorp seem to have same issue. Perhaps the guys over there will figure it out soon (if ever)

**Reply by suncokret12** [reply]
It does work in Gnome. Enter Picture-in-Picture, select your video, press Alt+Space to open context menu, and then just tick the 'Always on Top'

**Reply by o-data** [reply]
@suncokret12 it a Wayland issue from the look of it. Some sort of architecture/code system

**Reply by suncokret12** [reply]
It's not the Wayland issue; SmartTube devs can implement it to work out of the box.

**Reply by absidue** [reply]
@suncokret12 It is a wayland issue please reread my comment above: https://github.com/FreeTubeApp/FreeTube/discussions/5599#discussioncomment-10985846

Is there another SmartTube? because the one I've heard of is an Android TV app so I'm not sure how that is relevant to the discussion: https://github.com/yuliskov/smarttube

**Reply by o-data** [reply]
No points arguing with me :) . I'm just linking to what information is available regarding this topic, which seem very quiet...

So either there is some hidden bug, Wayland issue as posted in comments, or special workaround FreeTube don't have but SmartTube do?

I don't know... if you have a solution and able to get it fix, the Pull Request or #issue is the best place to be.

I posted in Discussion to see if it just me or there other out there like me because I simply can't find much mention of this issue. Thankfully there is not many but a few having the same issue.

There is one mention of this issue on Reddit, and 1-2 times in PopOS discord. But overall? A minority of people?

**Reply by suncokret12** [reply]
I wanted to say FreeTube, not SmartTube. My bad. Also, I'm not arguing. I said it's not a Wayland bug because they mentioned it's a security feature, not a bug. I said FreeTube devs can make it work because, for example, 'waybar' is always on top, including some other apps, so there must be a workaround.

**Reply by o-data** [reply]
If you technical enough you can post an issue or pull request. This is above my skill level honestly.

It been few months since I last test but I'm still hoping
